Paco-Server/src/com/google/sampling/experiential/server/CloudSQLDaoImpl.java

```diff
@@ -448,8 +452,13 @@ public JSONArray getResultSetAsJson(String query, Long eventId) throws SQLExcept
             //if client timezone, then do not write to json
             if (!(colName.equalsIgnoreCase(EventServerColumns.CLIENT_TIME_ZONE))) {
               //if date columns in projection, then display along with corresponding timezone
-              if ((colName.equalsIgnoreCase(EventServerColumns.RESPONSE_TIME)) || (colName.equalsIgnoreCase(EventServerColumns.SCHEDULE_TIME))) {
-                colValue = colValue + rs.getString(EventServerColumns.CLIENT_TIME_ZONE);
+              if(dateColumns.contains(colName)) {
+                offsetHrsStr =  rs.getString(EventServerColumns.CLIENT_TIME_ZONE);
+                offsetHrs = com.google.sampling.experiential.server.TimeUtil.getIntFromOffsetString(offsetHrsStr);
+                dateFromDb = new DateTime(rs.getTimestamp(colName).getTime());
+                // change the tz with value stored in db
+                dateInLocal = dateFromDb.withZoneRetainFields(DateTimeZone.forOffsetHours(offsetHrs));
```

Comment:
will the dateColumns also contain the 'when' column? If so, does that mean that this loop will also create a dateInLocal version of when and add it to the set of records in multipleRecords? It should not do that since when in local time is not a meaningful concept for any of our use cases.

Reply:
No, the date columns does not include when. It only includes response_time, schedule_time and sort_date